src/serialize/xml/normalize.ts

```diff
@@ -641,7 +642,7 @@ export class SWIDNormalizer extends BaseXmlNormalizer<Models.SWID> {
 
 export class ExternalReferenceNormalizer extends BaseXmlNormalizer<Models.ExternalReference> {
   normalize (data: Models.ExternalReference, options: NormalizerOptions, elementName: string): SimpleXml.Element | undefined {
```

Comment:
the encoding should have happened before the `XmlSchema.isAnyURI()` is run on it?
if it returns false, then `normalize` should return `undefined`

Reply:
done via https://github.com/CycloneDX/cyclonedx-javascript-library/pull/996/commits/95fb7c92e28f9bf5ed8f2378ad550f41d2f4d0bf